### PR TITLE
Deploy gyb in Swift 3 branch.

### PR DESCRIPTION
**Pull request by andersio**
One `Extensions.swift.gyb` template + three `[AppKit|UIKit|WatchKit]Extensions.gyb.py` configuration files.

### CODE REVIEW

File: Rex.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -395,6 +381,26 @@
...
+			path = WatchKit;
+			sourceTree = "<group>";
+		};
+		9AB3BCBA1D3D243C00063B96 /* Generated Sources */ = {
+			isa = PBXGroup;
+			children = (
+				9AB3BCB61D3D241A00063B96 /* AppKitExtensions.generated.swift */,
+				9AB3BCAE1D3D23E500063B96 /* WatchKitExtensions.generated.swift */,
+				9AB3BCB21D3D23FD00063B96 /* UIKitExtensions.iOS.generated.swift */,
+				9AB3BCB31D3D23FD00063B96 /* UIKitExtensions.tvOS.generated.swift */,
```
Comment (mdiep) [inline Rex.xcodeproj/project.pbxproj:398]: The best practice for generated source code files is to not add them to the project—but to generate them into `CONFIGURATION_BUILD_DIR` and use them from there.

File: Rex.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -867,6 +871,61 @@
...
+			buildActionMask = 2147483647;
+			files = (
+			);
+			inputPaths = (
+			);
+			outputPaths = (
+			);
+			runOnlyForDeploymentPostprocessing = 0;
+			shellPath = /bin/sh;
+			shellScript = "$PROJECT_DIR/script/gyb/gyb -Dframework=AppKit -Dtarget=macOS --line-directive '' -o $SOURCE_ROOT/Source/AppKit/AppKitExtensions.generated.swift $SOURCE_ROOT/Source/Extensions.swift.gyb";
```
Comment (mdiep) [inline Rex.xcodeproj/project.pbxproj:925]: Instead of using a script build phase, you should be able to create a `.gyb` build rule in the project.

**Review by mdiep — CHANGES_REQUESTED** [review]
Honestly, I'm not a huge fan of this approach. It does prevent some repetition, but it also adds some complexity and makes the project a little bit harder to debug and a little less approachable.

I'd rather focus on making the functions like `associatedProperty(_ host: AnyObject, keyPath: StaticString)` a little easier through the use of default values. I think we could probably use a default value of `#function` for some of the key properties, for instance.

**Comment by neilpa** [comment]
> Honestly, I'm not a huge fan of this approach. It does prevent some repetition, but it also adds some complexity and makes the project a little bit harder to debug and a little less approachable.

After seeing it in action I tend to agree. Maybe once the extensions are "complete" this approach will make more sense, but as-is I'm not inclined.

> I think we could probably use a default value of #function for some of the key properties, for instance.

I wen't through a few iterations trying to make this stuff less copy/paste-y but that's an idea I never considered.

**Comment by andersio** [comment]
In this case I think it would be better to focus on Swift 3 and the merge, and maybe readdress this or other alternatives in the merged RAC. :)

**Comment by mdiep** [comment]
Thanks @andersio!

**Pull request closed without merging**